File: examples/python/02_tpms_gyroid.py
#!/usr/bin/env python3
"""Example 02: Triply Periodic Minimal Surfaces (TPMS) Gyroid Infill Generation.

Demonstrates:
- Direct mathematical implicit surface slicing (Gyroid / Schwarz-P).
- Volumetric flow and process parameter control.
- G-code generation with native dry-core wasm/C libm engine.
"""
import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../../py/python"))
import dry

def main():
    print("=== Dry Example 02: TPMS Gyroid Infill ===")

    options = {
        "surface": "gyroid",
        "cellSize": 15.0,
        "cellsX": 2,
        "cellsY": 2,
        "cellsZ": 2,
        "isovalue": 0.0,
        "resolution": 40,
        "sliceSpacing": 0.25,
        "wallThickness": 0.45,
    }

    print(f"Generating TPMS '{options['surface']}' cellular lattice:")
    print(f"  Cell Size: {options['cellSize']} mm, Grid: {options['cellsX']}x{options['cellsY']}x{options['cellsZ']}")

    # 1. Generate G-code directly via engine
    gcode = dry.tpms_gcode(options)
    print(f"✓ Generated {len(gcode)} lines of G-code.")

    # 2. Inspect first few lines
    print("--- Sample G-code lines ---")
    for line in gcode[:8]:
        print("  ", line)

if __name__ == "__main__":
    main()

File: examples/typescript/02_tpms_gyroid.ts
/**
 * Dry TypeScript Example 02: TPMS Cellular Gyroid Lattice Infill
 */
import { tpms_gcode, tpms_ops } from '@dry/sdk';

function main() {
  console.log("=== Dry TypeScript Example 02: TPMS Gyroid Lattice ===");

  const options = {
    surface: "gyroid",
    cell_size: 15.0,
    grid_size: [2, 2, 2] as [number, number, number],
    layer_height: 0.3,
    iso_level: 0.0,
    stepover: 0.6,
  };

  const ops = tpms_ops(options);
  console.log(`✓ Generated ${ops.length} TPMS L1 toolpath operations.`);

  const gcode = tpms_gcode(options);
  console.log(`✓ Emitted ${gcode.split('\n').length} lines of machine-ready G-code.`);
}

main();
